pkg/segment/structs/evaluationstructs.go

```diff
@@ -359,17 +362,19 @@ func (self *SortExpr) ReleaseProcessedSegmentsLock() {
 
 func findNullFields(fields []string, fieldToValue map[string]utils.CValueEnclosure) ([]string, error) {
 	nullFields := []string{}
+	var err error
 	for _, field := range fields {
 		val, exists := fieldToValue[field]
 		if !exists {
-			return []string{}, fmt.Errorf("findNullFields: Expression has a field for which value is not present")
+			err = fmt.Errorf("findNullFields: Expression has a field %v for which value is not present", field)
+			continue
```

Comment:
What's the purpose of this change? Why don't we want to return on the first error?

Reply:
I want to know the null fields and evaluate those to false. There might be some cases where we might want to know which are the null fields irrespective or if a field that was supposed to have a value does not have it. This function was early exiting so I think this would be a better approach.

Reply:
addressed in latest commit